### PR TITLE
Add documentation for Widget Mode configuration

### PR DESCRIPTION
**Pull request by greenberga**
image

---

image

### CODE REVIEW

**Comment by mikemherron** [comment]
Really nicely explained 👍 I think it's great. One minor suggestion, "the widget can complete automatically, with no user interaction" -> what about "the widget will complete automatically..." ?

I was going to suggest being explicit that this is different from v1... but maybe that would just add to the confusion? I don't have a strong opinion.

File: docs/sdk/configuration.md [hunk 1]
```diff
@@ -83,3 +81,23 @@ Example:
 
 ## Javascript SDK
 See the [**CreateWidgetOptions**](./reference/sdk.createwidgetoptions.md) for the settings you can pass when creating a widget using `sdk.createWidget()`.
+
+## Widget Mode
+
+The point at which a widget can *begin* solving challenges depends on its [start mode](#data-start-attribute). Whether or not the widget can automatically *complete* depends on a different configuration option: the Widget Mode.
```
Comment (gzuidhof) [inline docs/sdk/configuration.md:87]:
```suggestion
The point at which a widget can *begin* solving challenges depends on its [*start mode*](#data-start-attribute). Whether or not the widget can automatically *complete* depends on a different configuration option: the Widget Mode.
```

**Review by mikemherron — APPROVED** [review]

File: docs/sdk/configuration.md [hunk 2]
```diff
@@ -83,3 +81,23 @@ Example:
...
+
+A widget can be configured with one of 3 different Widget Modes:
+
+- **One-click:** a user must click once on the widget in order for it to complete.
+- **Zero-click:** the widget can complete automatically, with no user interaction.
+- **Smart (default):** the widget intelligently decides, based on signals from the current browsing session, whether or not a click is needed to complete.
+
+There is a trade-off between requiring a click or not. A widget that completes with zero user interaction presents the least friction. But requiring just a single click greatly enriches the signals that can be used to identify bots. For that reason, we recommend sticking with the default, which is to allow the widget to decide when to require a click.
```
Comment (gzuidhof) [inline docs/sdk/configuration.md:95]:
```suggestion
There is a trade-off between requiring a click or not. A widget that completes without user interaction presents the least friction, but requiring just a single click allows us to better identify bots. For that reason, we recommend sticking with the default, which is to allow the widget to decide when to require a click.
```

File: docs/sdk/configuration.md [hunk 3]
```diff
@@ -83,3 +81,23 @@ Example:
...
+
+:::tip[no traffic lights here]
+
+There will never be tedious tasks like clicking images of traffic lights. One single click is the only required interaction.
```
Comment (gzuidhof) [inline docs/sdk/configuration.md:99]:
```suggestion
There are never any tedious tasks like clicking images of traffic lights. One single click is the only required interaction.
```

(If you think it's better)

File: docs/sdk/configuration.md [hunk 4]
```diff
@@ -83,3 +81,23 @@ Example:
...
+
+:::
+
+The Widget Mode is configured per *application*, rather than per widget. If it were configurable in the widget (the way start mode is configured), bad actors could simply toggle it to their advantage. You can set the Widget Mode when creating or managing an application from within the dashboard.
```
Comment (gzuidhof) [inline docs/sdk/configuration.md:103]:
```suggestion
The Widget Mode is configured per *application* in the [Friendly Captch dashboard](https://app.friendlycaptcha.eu). If this were configured on your website, an attacker could change this setting to their advantage.
```

**Review by gzuidhof — APPROVED** [review]
LGTM :) Added some suggestions (feel free to ignore the ones you disagree with)

**Pull request merged**